webapps-schema/src/main/java/io/camunda/webapps/schema/descriptors/AbstractIndexDescriptor.java

```diff
@@ -26,7 +26,19 @@ public AbstractIndexDescriptor(final String indexPrefix, final boolean isElastic
 
   @Override
   public String getFullQualifiedName() {
-    return String.format("%s-%s-%s_", getIndexPrefix(), getIndexName(), getVersion());
+
+    //    legacy support for descriptors in operate where the index prefix contains the component
+    // name e.g. indexPrefix = "operate"
+    if (getIndexPrefix() != null && getIndexPrefix().contains(getComponentName())) {
+      return String.format("%s-%s-%s_", getIndexPrefix(), getIndexName(), getVersion());
+    }
+
+    if (getIndexPrefix() == null || getIndexPrefix().isEmpty()) {
+      return String.format("%s-%s-%s_", getComponentName(), getIndexName(), getVersion());
+    } else {
+      return String.format(
+          "%s-%s-%s-%s_", getIndexPrefix(), getComponentName(), getIndexName(), getVersion());
+    }
```

Comment:
Couldn't we just do:

```suggestion
      return String.format(
          "%s-%s-%s-%s_", getIndexPrefix(), getComponentName(), getIndexName(), getVersion());
```

Reply:
for legacy descriptors in operate the index prefix will contain the string `operate` or `tasklist` causing the indices to be for example `prefix-operate-operate-index-8.x.x` and if we provide an empty prefix we will have our names start with a `-` e.g. `-operate-index-8.x.x`

Reply:
So like?
```
var prefix = isBlank() ? getComponent() : getIndexPrefix() + "-" + getComponent();
return String.format("%s-%s-%s_", prefix, getComponentName(), getIndexName(), getVersion());
```

Reply:
this would break for legacy descriptors where 
prefix = operate
component name = operate
index name = index
verison = 8.0.0

it would return operate-operate-index-8.0.0, take a look at `OperationTemplate.java` as an example